src/hydra/dataProvider.js

```diff
@@ -360,38 +359,44 @@ export default (
    *
    * @returns {Promise}
    */
-  const fetchApi = (type, resource, params) => {
+  const fetchApi = (type, resource, params) =>
+    convertReactAdminRequestToHydraRequest(type, resource, params)
+      .then(({url, options}) => httpClient(url, options))
+      .then(response =>
+        convertHydraResponseToReactAdminResponse(type, resource, response),
+      );
+
+  return {
+    getList: (resource, params) => fetchApi(GET_LIST, resource, params),
+    getOne: (resource, params) => fetchApi(GET_ONE, resource, params),
     // Hydra doesn't handle MANY requests, so we fallback to calling the ONE request n times instead
-    switch (type) {
-      case 'INTROSPECT':
-        if (apiSchema) return Promise.resolve({data: apiSchema});
-        return apiDocumentationParser(entrypoint).then(({api}) => {
-          apiSchema = api;
-          return {data: api};
-        });
-
-      case GET_MANY:
-        return Promise.all(
-          params.ids.map(id =>
-            reactAdminDocumentsCache[id]
-              ? Promise.resolve({data: reactAdminDocumentsCache[id]})
-              : fetchApi(GET_ONE, resource, {id}),
-          ),
-        ).then(responses => ({data: responses.map(({data}) => data)}));
-
-      case DELETE_MANY:
-        return Promise.all(
-          params.ids.map(id => fetchApi(DELETE, resource, {id})),
-        ).then(responses => ({data: []}));
-
-      default:
-        return convertReactAdminRequestToHydraRequest(type, resource, params)
-          .then(({url, options}) => httpClient(url, options))
-          .then(response =>
-            convertHydraResponseToReactAdminResponse(type, resource, response),
-          );
-    }
+    getMany: (resource, params) =>
+      Promise.all(
+        params.ids.map(id =>
+          reactAdminDocumentsCache[id]
```

Comment:
I'm not sure what that is, but react-admin already uses an internal cache on the client-side. If the dataProvider adds another layer of caching, this can cause stale data.

Also, all other methods call `fetchAPI` directly, why should this one use a cache and not the others?

Reply:
It has been done like this since a long time. Maybe @dunglas could answer this better than me?